controllers/progressivesync_controller.go

```diff
@@ -388,32 +396,34 @@ func (r *ProgressiveSyncReconciler) reconcileStage(ctx context.Context, ps syncv
 		r.updateStageStatus(ctx, stage.Name, message, syncv1alpha1.PhaseFailed, &ps)
 		failed := ps.NewStatusCondition(syncv1alpha1.CompletedCondition, metav1.ConditionFalse, syncv1alpha1.StagesFailedReason, message)
 		apimeta.SetStatusCondition(ps.GetStatusConditions(), failed)
-		return ps, ctrl.Result{}, err
+		return ps, ctrl.Result{RequeueAfter: requeueDelayOnError}, err
 	}
-	log.Info("apps selected", "apps", utils.GetAppsName(apps))
+	log.Info("fetched apps targeting selected clusters", "apps", utils.GetAppsName(apps))
 
-	// Remove the annotation from the OutOfSync Applications before passing them to the Scheduler
+	// Remove the annotation from the Healthy and OutOfSync Apps before passing them to the Scheduler.
+	// Healthy and OutOfSync apps are applications that have not been synced at this run of the Progressive Sync.
 	// This action allows the Scheduler to keep track at which stage an Application has been synced.
-	outOfSyncApps := utils.GetAppsBySyncStatusCode(apps, argov1alpha1.SyncStatusCodeOutOfSync)
-	if err := r.removeAnnotationFromApps(ctx, outOfSyncApps, utils.ProgressiveSyncSyncedAtStageKey); err != nil {
-		message := "failed to remove out-of-sync annotation from apps"
-		log.Error(err, message)
-
-		r.updateStageStatus(ctx, stage.Name, message, syncv1alpha1.PhaseFailed, &ps)
-		// Set ProgressiveSync status
-		failed := ps.NewStatusCondition(syncv1alpha1.CompletedCondition, metav1.ConditionFalse, syncv1alpha1.StagesFailedReason, message)
-		apimeta.SetStatusCondition(ps.GetStatusConditions(), failed)
-
-		return ps, ctrl.Result{}, err
-	}
+	//outOfSyncApps := utils.GetAppsBySyncStatusCode(apps, argov1alpha1.SyncStatusCodeOutOfSync)
+	//healthyApps := utils.GetAppsByHealthStatusCode(outOfSyncApps, health.HealthStatusHealthy)
```

Comment:
This condition needs to be `not Progressing` as we want to give a bad deployment time to recovery